src/actions/exportToCSV/ExportToCSVAction.js

```diff
@@ -47,49 +47,49 @@ export default class ExportToCSVAction {
         }
 
         const object = objectPath[0];
-        const hasChildren = object.composition && object.composition.length;
+        const composition = this.openmct.composition.get(object);
 
-        //allow if the object has children or if it is a telemetry object
-        return hasChildren || SUPPORTED_TYPES.includes(object.type);
+        //allow if the object has composition or if it is a telemetry object
+        return composition || SUPPORTED_TYPES.includes(object.type);
     }
 
     // Exports telemetry (or groups of telemetry) data into CSV file
     async invoke(objectPath) {
         const object = objectPath[0];
         let parameterIds = [];
-        let objectPromises = [];
+        let parameterIdsPromiseResolve;
+        let parameterIdsPromise = new Promise((resolve) => {
```

Comment:
I _think_ we can simplify the Promises handling in this function (though I may have missed something).

```JavaScript
let parameterIdsPromise;
// ...
if (composition) {
  parameterIdsPromise = composition.load()
  //...
} else {
  parameterIds.push(this.getParameter(object.identifier));
  parameterIdsPromise = Promise.resolve(parameterIds);
}
//...
parameterIdsPromise.then(async () => {
//...
```

Reply:
Hmm composition.load() returns objects though not ids, so I did use your idea with some modifications. Hope that works.